Essentials/src/com/earth2me/essentials/Teleport.java

```diff
@@ -247,6 +249,11 @@ protected void respawnNow(IUser teleportee, TeleportCause cause) throws Exceptio
     //The warp function is a wrapper used to teleportPlayer a player to a /warp
     @Override
     public void warp(IUser teleportee, String warp, Trade chargeFor, TeleportCause cause) throws Exception {
+        EssentialsWarpEvent event = new EssentialsWarpEvent(teleportee, warp);
+        Bukkit.getServer().getPluginManager().callEvent(event);
+        if(event.isCancelled()){
```

Comment:
This should be spaced out.

Reply:
Testing with Remote Debugging and also Checkpoints, the event was called. I haven't test the listeners tho